bundles/org.openhab.binding.energenie/src/main/java/org/openhab/binding/energenie/internal/handler/EnergenieSocket.java

```diff
@@ -0,0 +1,186 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.energenie.internal.handler;
+
+import static org.openhab.binding.energenie.internal.EnergenieBindingConstants.*;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.Socket;
+import java.net.UnknownHostException;
+import java.nio.ByteBuffer;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.smarthome.core.util.HexUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Class handling the Socket connections.
+ *
+ * @author Hans-Jörg Merk - Initial contribution
+ * @author Hilbrand Bouwkamp - Moved Socket to it's own class
+ */
+@NonNullByDefault
+class EnergenieSocket {
+
+    private static final int SOCKET_TIMEOUT_MILLISECONDS = 1500;
+    private static final byte[] MESSAGE = { 0x11 };
+
+    private final Logger logger = LoggerFactory.getLogger(EnergenieSocket.class);
+    private final String host;
+    private final byte[] key;
+
+    public EnergenieSocket(final String host, final String password) {
+        this.host = host;
+        key = getKey(password);
+    }
+
+    private byte[] getKey(final String password) {
+        final int passwordLength = password.length();
+        String passwordString = password;
+        for (int i = 0; i < (8 - passwordLength); i++) {
+            passwordString = passwordString + " ";
+        }
+        return passwordString.getBytes();
```

Comment:
The default charset varies from system to system so it is better to specify it yourself instead of using the default.

Reply:
Did not see such an implementation in other bindings reading passwords from thing config so don't understand why it should be used here.

Reply:
Well usually the default charset is ascii based so it normally isn't a problem in western countries. But try running this on a chinese or arabic computer and you get some interesting results. The reason that you don't see it done in other bindings is because those developers were either unaware or didn't care about this issue since it wouldn't affect the vast majority of environments.

Reply:
Actually now that I think about it maybe using the default locale charset is the correct one to use. LOL. Disregard my change suggestion then!